features/features.go

```diff
@@ -0,0 +1,63 @@
+package features
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/earthly/earthly/ast/spec"
+	"github.com/earthly/earthly/util/flagutil"
+)
+
+// Features is used to denote which features to flip on or off; this is for use in maintaining
+// backwards compatibility
+type Features struct {
+	// DoSaves flags the feature to only save artifacts under directly referenced targets
+	// of those referenced by BUILDs
+	ReferencedSaveOnly bool `long:"referenced-save-only" description:"only save artifacts that are directly referenced"`
+}
+
+var errUnexpectedArgs = fmt.Errorf("unexpected VERSION arguments; should be VERSION [flags] <major-version>.<minor-version>")
+
+// GetFeatures returns a features struct for a particular version
+func GetFeatures(version *spec.Version) (*Features, error) {
+	var ftrs Features
+
+	if version == nil {
+		return &ftrs, nil
+	}
+
+	if version.Args == nil {
+		return nil, errUnexpectedArgs
+	}
+
+	parsedArgs, err := flagutil.ParseArgs("VERSION", &ftrs, version.Args)
```

Comment:
Could this be made to ignore feature flags that we don't recognize? Maybe only show a warning in such cases?

Reply:
I'm unsure if issuing a warning is a good idea here, 

What about a case where a user wants to make use of a certain feature but that feature doesn't exist? for example:

    VERSION --enable-foo 0.5

If we ran this on an earlier version of earthly, then it would just print a warning and then likely fail later on when foo isn't enabled.

Can you think of an example where this ignore unknown flags behaviour would help a user?

Reply:
Not a strong opinion. I was thinking of a case where multiple people on the team are using varying versions of Earthly and only some have the version with that feature flag. If the feature flag errors out on a version of Earthly that doesn't support it, then the users wouldn't be able to use any target from that Earthfile.

I'm not convinced either that this behavior is useful. So maybe it's better to return an error if we don't recognize it.

Reply:
Maybe that's where `--strict` could come into play? I'm in favour of leaving it as-is until a use-case comes up requiring it.

Reply:
Yeah, less is more. As-is is best.